### PR TITLE
[Ink] New bounded ink should ignore maxRippleRadius property

### LINKED ISSUE #329: Bounded ink should ignore the maxRippleRadius property
Right now it's possible to have bounded ink (should fill the view) with a smaller maxRippleRadius and the visual result looks like unbounded ink (a circle). 
1. Create an MDCFlatButton
2. Set .inkMaxRippleRadius to some smaller value.
3. Result: Circular ink, not filling the view.

In MDCInk, just ignore maxRippleRadius if the style is unbounded.

### PR DESCRIPTION
**Pull request by romoore**
For the new Ink, when the view uses `.bounded` for `inkStyle`, it should
ignore the `maxRippleRadius` property. No changes should be made to legacy Ink
since we're migrating away from it and we don't want to make unnecessary
changes to existing clients.

Closes #329

### CODE REVIEW

File: components/Ink/src/MDCInkView.m
Comment (ianegordon) [inline components/Ink/src/MDCInkView.m]: Don't we need to check for inkStyle before we set maxRippleRadius below?
Otherwise what happens if we setInkStyle:Unbounded, then setRadius:12 ?

**Review by ianegordon — DISMISSED** [review]

File: components/Ink/src/MDCInkView.m
Comment (ajsecord) [inline components/Ink/src/MDCInkView.m]: Sort capitals before lowercase.

File: components/Ink/src/MDCInkView.m
Comment (yarneo) [reply components/Ink/src/MDCInkView.m]: Also what happens if `maxRippleRadius` is set to a number initially, but because it is bounded `maxRippleRadius` is 0 (because default is Bounded). Then someone changes the radius to 0 explicitly and makes the ink Unbounded. The number retained in `_maxRippleRadius` will show the previous number and not 0, because it doesn't pass the `MDCCGFloatEqual` conditional above.

Comment (romoore) [reply components/Ink/src/MDCInkView.m]: Done.

File: components/Ink/src/MDCInkView.m
Comment (yarneo) [inline components/Ink/src/MDCInkView.m]: is it more accurate to say here if it isn't Bounded rather than if it's Unbounded? Looks like we want to update maxRippleRadius only when inkStyle is not Bounded?

File: components/Ink/src/MDCInkView.m
Comment (yarneo) [inline components/Ink/src/MDCInkView.m]: aren't we losing the original maxRippleRadius here if we set it to 0 when bounded? so when it is set to unbounded we will get 0 and not the original value

File: components/Ink/src/MDCInkView.m
Comment (romoore) [reply components/Ink/src/MDCInkView.m]: The style is `.unbounded` so I'm trying to work with that terminology. I'm not sure how negating the value of an `NS_ENUM` would make things clearer. Would you prefer to use a `switch` statement instead?

Do you have a suggestion you could put in a snippet?

File: components/Ink/src/MDCInkView.m
Comment (romoore) [reply components/Ink/src/MDCInkView.m]: I'm not sure I understand. The ivar was not previously being used and now it's only used to "restore" the assigned value when switching to `.unbounded` ink.

File: components/Ink/src/MDCInkView.m
Comment (yarneo) [reply components/Ink/src/MDCInkView.m]: From what I understand we want the `maxRippleRadius` to change only in the case where it is not bounded right? Because bounded it needs to be set to 0 no matter what. In that case all other options should allow to have the `maxRippleRadius` to be set, and not only .unbounded. As an example if we add more `inkStyle`s in the future

File: components/Ink/src/MDCInkView.m
Comment (yarneo) [reply components/Ink/src/MDCInkView.m]: Just tested my hypothesis, this seems fine, please ignore.

File: components/Ink/src/MDCInkView.m
Comment (romoore) [reply components/Ink/src/MDCInkView.m]: I've refactored to use a `switch`. Please take a look.

File: components/Ink/src/MDCInkView.m
Comment (ianegordon) [inline components/Ink/src/MDCInkView.m]: Why does this require a setNeedsLayout?
If we need one here, do we need one where we set self.inkLayer.maxRippleRadius above?

**Review by ianegordon — DISMISSED** [review]

**Review by yarneo — APPROVED** [review]
LGTM % Ian's comment

File: components/Ink/src/MDCInkView.m
Comment (romoore) [reply components/Ink/src/MDCInkView.m]: Done. It's needed for legacy Ink but not new Ink.

**Review by ajsecord — COMMENTED** [review]
Drive by comment!

File: components/Ink/src/MDCInkView.m
Comment (romoore) [reply components/Ink/src/MDCInkView.m]: Gah! You caught me! 😨 I was trying to sneak it in...

**Review by ianegordon — APPROVED** [review]
LGTM, pending a +1 from Travis.

**Comment by ianegordon** [comment]
Would you kick off another TravisCI integration?

**Comment by romoore** [comment]
Travis is failing to find an iOS11/iPhoneX environment:

```
Failed to test MDCUnitTests in catalog/MDCCatalog.xcworkspace.
Build settings from command line:
    GCC_GENERATE_TEST_COVERAGE_FILES = YES
    GCC_INSTRUMENT_PROGRAM_FLOW_ARCS = YES
xcodebuild: error: Unable to find a destination matching the provided destination specifier:
		{ platform:iOS Simulator, OS:11.0, name:iPhone X }
	The requested device could not be found because no available devices matched the request.

```

In the logs, I don't see any iOS 11.x being listed.

**Comment by yarneo** [comment]
did you restart the job via travis, or did you start a new job? i suspect maybe when restarting a job it might cache certain things, like using the wrong xcode (which was the problem during last weekend)

**Comment by romoore** [comment]
I did neither, I clicked "Restart build
screen shot 2018-02-26 at 2 58 17 pm

**Comment by romoore** [comment]
Travis passed, merging.

**Pull request merged**
